servicetalk-concurrent-test-internal/src/main/java/io/servicetalk/concurrent/test/internal/Utils.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.test.internal;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+public final class Utils {
+    private Utils() {
+        // no instances
+    }
+
+    public static void awaitUninterruptibly(CountDownLatch latch) {
+        boolean interrupted = false;
+        try {
+            do {
+                try {
+                    latch.await();
+                    return;
+                } catch (InterruptedException e) {
+                    interrupted = true;
+                }
+            } while (true);
+        } finally {
+            if (interrupted) {
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    public static boolean awaitUninterruptibly(CountDownLatch latch, long timeout, TimeUnit unit) {
+        final long startTime = System.nanoTime();
+        final long timeoutNanos = NANOSECONDS.convert(timeout, unit);
+        long waitTime = timeoutNanos;
+        boolean interrupted = false;
+        try {
+            do {
+                try {
+                    return latch.await(waitTime, NANOSECONDS);
+                } catch (InterruptedException e) {
+                    interrupted = true;
+                }
+                waitTime = timeoutNanos - (System.nanoTime() - startTime);
+                if (waitTime <= 0) {
+                    return true;
+                }
+            } while (true);
+        } finally {
+            if (interrupted) {
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    public static <T> T takeUninterruptibly(BlockingQueue<T> queue) {
```

Comment:
`takeUninterruptibly` and `pollUninterruptibly` is not used by any other code, except `TestPublisherSubscriber`. Can we keep them there for now and share only when necessary?

Reply:
It seemed re-use would be more likely by exposing them. This is still only in the test-internal module so not part of the public API.

Is the motivation for keeping them local based on YAGNI, the behaviour of the methods (which some may disagree with), or other reasons?

Reply:
The only motivation was to not move code until necessary to minimize git changes, but either way works fine for me in this case.